### ISSUE TITLE
Add mark filtering to context anchor

### ISSUE THREAD

**Issue by MariannaPaszkowska** [googlefonts/glyphsLib#910]

**Comment by anthrotype** [comment]
are you going to provide a bit more context to understand what this is about? thanks :)

**Comment by schriftgestalt** [comment]
The anchor context can now have a lookup flag prefix. 
When you could do `C *` before, now you can do: 
```
lookupflag UseMarkFilteringSet @CombiningTopAccents;C *
```
Everything before the semicolon is supposed to be at the beginning of the lookup. I do a dict lookup with the prefix to collect all rules with the same prefix.

**Comment by simoncozens** [comment]
Even without the mark filtering, we don't currently support contextual anchors at all - but I'm pretty sure we will by the end of today...

**Comment by schriftgestalt** [comment]
I thought @khaledhosny had worked on it?

**Comment by khaledhosny** [comment]
No, I didn’t do any work related to contextual anchors.

**Comment by simoncozens** [comment]
I've done 2 and chucked it into a branch (https://github.com/googlefonts/glyphsLib/tree/contextual-mark-writer) so I can use it. It's not ready for PR yet as it doesn't handle the mkmk case.

**Comment by khaledhosny** [comment]
My vote for 2) as well, and it can be also extended to ignore anchors that Glyphs ignores (IIRC any anchors starting with non-letter, other than `*` and `_` of course).

**Comment by anthrotype** [comment]
> set the UFO feature writer lib key to use it

note that setting that lib key overwrites the default list of writers, which means if you still want the default ones to be run you'll have to list them explicitly in the list. We added an `...` placeholder to mean the default writers, but that only works when you use the fontmake CLI option --feature-writers or when you call ufo2ft compile methods and pass the `featureWriters` parameter.

**Comment by simoncozens** [comment]
We have a `DEFAULT_FEATURE_WRITERS` constant in glyphsLib constants, which we check when round-tripping. So I did this: https://github.com/googlefonts/glyphsLib/commit/7dacee459a3a099c8a4ce31327c9bc04111d4817

**Comment by anthrotype** [comment]
yeah but we stopped using that DEFAULT_FEATURE_WRITERS at some point because it was outdated, ufo2ft gained another default writer I belive (gdefFeatureWriter?) and if we hard-coded the list we wouldn't get that

**Comment by simoncozens** [comment]
Or we can grab it from `ufo2ft.featureCompiler.FeatureCompiler.defaultFeatureWriters`.

**Comment by anthrotype** [comment]
can we not change the ufo2ft feature writer instead so both glyphs and ufo will gain from this?

**Comment by simoncozens** [comment]
My point is we _don't_ want to keep it updated with ufo2ft, in the sense that we don't want to update our list of feature writers just because ufo2ft has. If ufo2ft does something new, we _only_ want to implement that if it's something that Glyphs also does.
